src/analysis/economics.jl

```diff
@@ -0,0 +1,169 @@
+    seeks to maximise outputs for a given level of input, or input oriented (`orient=:Input`),
+    which seeks to minimise an input for a given level of output.
+
+# Returns
+DEAResult, which summarizes inputs, outputs, efficiencies and peers for each scenario.
+
+# Examples
+```julia
+dom = ADRIA.load_domain("example_domain")
+scens = ADRIA.sample(dom, 128)
+rs = ADRIA.run_scenarios(dom, scens, "45")
+
+# Get cost of deploying corals in each scenario, with user-specified function
+cost = cost_function(scens)
+
+# Get mean coral cover and shelter volume for each scenario
+s_tac = dropdims(
+    mean(ADRIA.metrics.scenario_total_cover(rs); dims=:timesteps); dims=:timesteps
+)
+s_sv::Vector{Float64} =
+    dropdims(
+        mean(mean(ADRIA.metrics.absolute_shelter_volume(rs); dims=:timesteps); dims=:sites);
+        dims=(:timesteps, :sites)
+    )
+
+# Do output oriented DEA analysis seeking to maximise cover and shelter volume for minimum
+# deployment cost.
+DEA_scens = ADRIA.economics.data_envelopment_analysis(cost, s_tac, s_sv)
+
+```
+
+# References
+1. Huguenin, J-M., 2012.
+   Data Envelopment Analysis (DEA): A pedagogical guide for decision makers in the public
+   sector. https://api.semanticscholar.org/CorpusID:188267263
+2. Pascoe, S., Cannard, T., Dowling, N.A., et. al., 2023.
+   Use of Data Envelopment Analysis (DEA) to assess management alternatives in the presence
+   of multiple objectives.
+   Marine Policy, 148, 105444.
+   https://doi.org/10.1016/j.marpol.2022.105444
+3. Pascoe, S., 2024.
+   On the use of Data Envelopment Analysis for Multi-Criteria Decision Analysis.
+   Algorithms, 17:89.
+   https://doi.org/10.3390/a17030089
+
+"""
+function data_envelopment_analysis(rs::ResultSet, input_function::Function,
+    metrics...; orient::Symbol=:Output, dea_model::Function=DEA.deabigdata
+)::DEAResult
+    X = input_function(rs.inputs)
+    return data_envelopment_analysis(
+        X, metrics; orient=orient, dea_model=dea_model
+    )
+end
+function data_envelopment_analysis(rs::ResultSet, Y::YAXArray, input_function::Function;
+    orient::Symbol=:Output, dea_model::Function=DEA.deabigdata
+)::DEAResult
+    X = input_function(rs.inputs)
+    return data_envelopment_analysis(
+        X, Y; orient=orient, dea_model=dea_model
+    )
+end
+function data_envelopment_analysis(
+    X::YAXArray, metrics...; orient::Symbol=:Output, dea_model::Function=DEA.deabigdata
+)::DEAResult
+    Y = Array(hcat(metrics...))
+    return data_envelopment_analysis(
+        Array(X), Y; orient=orient, dea_model=dea_model
+    )
+end
+function data_envelopment_analysis(
+    X::YAXArray, Y::YAXArray; orient::Symbol=:Output, dea_model::Function=DEA.deabigdata
+)::DEAResult
+    return data_envelopment_analysis(
+        Array(X), Array(Y); orient=orient, dea_model=dea_model
+    )
+end
+function data_envelopment_analysis(
+    X::Vector{Float64}, Y::Matrix{Float64}; orient::Symbol=:Output,
```

Comment:
All the wrapper functions allow X and Y to be input as YAXArrays, so the user can use a YAXArray as input without having to convert to vector or Array. I can change the inner function to AbstractArray inputs however, although the function will error if the inputs are not Vector or Matrix type because this is what the DEA model takes. Let me know how you'd like me to proceed.

Reply:
Sorry, about my second comment, I think I was just wrong. 

But about my first comment: if you want to call `data_envelopment_analysis` passing `(rs, Y, input_function)`, `Y` needs to be a YAXArray (so, if the user has an Array it won't work). Same thing with the method that accepts only `X` as argument. But that's a question for you: do you want to give support for versions of those functions with `Array`s instead of `YAXArray`s? If that's not an issue, then maybe you don't need to change anything..?

Reply:
Ah ok I see what you mean, thanks! I think I'll allow Y to be an AbstractArray because this could be calculated with any "cost" function, but X should be a YAXAarry because it'll be the result of an ADRIA metric